sqlmesh/core/snapshot/definition.py

```diff
@@ -927,6 +935,26 @@ def to_end_date(end: int, unit: IntervalUnit) -> TimeLike:
             else default
         )
 
+    def needs_destructive_check(
+        self,
+        allow_destructive_snapshots: t.Set[str],
+        is_forward_only_plan: bool = False,
+        runtime_check: bool = False,
+    ) -> bool:
+        return (
+            self.is_model
+            # is a forward-only change happening?
```

Comment:
I think this method is too overloaded and hard to digest. We should only keep the logic that is shared between plantime and runtime here and do plan / run specific stuff upstream.